pkg/template/resource.go

```diff
@@ -85,3 +91,14 @@ func applyParamToResourceTemplate(param pipelinev1.Param, rt json.RawMessage) js
 	paramVariable := fmt.Sprintf("$(params.%s)", param.Name)
 	return []byte(strings.Replace(string(rt), paramVariable, param.Value.StringVal, -1))
 }
+
+// Uid generates a postfix like the k8s apiserver does for the generateName field.
+func Uid() string {
+	return rand.String(5)
+}
+
+// ApplyUIDToResourceTemplate returns the TriggerResourceTemplate after uid replacement
+// The same uid should be used per trigger to properly address resources throughout the TriggerTemplate.
+func ApplyUIDToResourceTemplate(rt json.RawMessage, uid string) json.RawMessage {
+	return bytes.Replace(rt, uidMatch, []byte(uid), -1)
```

Comment:
nit: since we're using `bytes.Replace()` here, can you please change the above
```
return []byte(strings.Replace(string(rt), paramVariable, param.Value.StringVal, -1))
```
to also use `bytes.replace()` for consistency?